synth.py

```diff
@@ -2,61 +2,68 @@
 import synthtool.gcp as gcp
 import logging
 import subprocess
+import json
 import os
 
 logging.basicConfig(level=logging.DEBUG)
 
-AUTOSYNTH_MULTIPLE_COMMITS = True
+gapic = gcp.GAPICMicrogenerator()
 
-
-gapic = gcp.GAPICGenerator()
 common_templates = gcp.CommonTemplates()
 
 # tasks has two product names, and a poorly named artman yaml
 version = 'v1'
-library = gapic.node_library(
-    'pubsub', version, config_path="/google/pubsub/artman_pubsub.yaml")
+library = gapic.typescript_library(
+    'pubsub',
+    version,
+    generator_args={
+        'grpc-service-config': f'google/pubsub/{version}/pubsub_grpc_service_config.json',
+        'package-name': f'@google-cloud/pubsub',
+        'main-service': f'pubsub',
+        'bundle-config': f'google/pubsub/{version}/pubsub_gapic.yaml',
+        'template': f'typescript_gapic'
+    },
+    proto_path=f'/google/pubsub/{version}',
+    extra_proto_files=['google/cloud/common_resources.proto']
+)
 
 # skip index, protos, package.json, and README.md
 s.copy(
     library,
-    excludes=['package.json', 'README.md', 'src/index.js'])
+    excludes=['package.json', 'README.md', 'src/index.ts'])
 
 templates = common_templates.node_library(source_location='build/src')
 s.copy(templates)
 
-# https://github.com/googleapis/gapic-generator/issues/2127
-s.replace("src/v1/subscriber_client.js",
-          "  }\n\s*/\*\*\n\s+\* The DNS address for this API service\.",
-          "\n    // note: editing generated code\n"
-          "    this.waitForReady = function(deadline, callback) {\n"
-          "      return subscriberStub.then(\n"
-          "        stub => stub.waitForReady(deadline, callback),\n"
-          "        callback\n"
-          "      );\n"
-          "    };\n"
-          "    this.getSubscriberStub = function() {\n"
-          "      return subscriberStub;\n"
-          "    };\n"
-          "\g<0>")
+# surgery in client.ts file
+clients = ['publisher', 'subscriber']
+for client_name in clients:
+    client_file = f'src/v1/{client_name}_client.ts'
+    s.replace(client_file,
```

Comment:
Can we please reference tracking bugs where we're rationalizing what's happening here?

Reply:
Sure. I will add comment.
FYI, @xiaozhenliu-gg5 is working on put common IAM service in gax [PR] (https://github.com/googleapis/gax-nodejs/pull/762/files)
It will get rid of these hack once the PR merged.

Reply:
Oh hey, this is great!

Reply:
👍fix will be there after Node 8 deprecation, and this surgery part will go away. Thanks!